src/modules/assets/assets.entity.ts

```diff
@@ -0,0 +1,37 @@
+import { Field, Int, ObjectType } from '@nestjs/graphql';
+import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+import { DIDEntity } from '../did/did.entity';
+
+@Entity()
+@ObjectType()
+export class Asset {
+  static create(data: Partial<Asset>) {
+    const entity = new Asset();
+    Object.assign(entity, data);
+    return entity;
+  }
+  @PrimaryColumn({ type: 'text' })
+  @Field(() => Int)
+  id: string;
+
+  @Column()
+  @Field()
+  owner: string;
+
+  @Column({ nullable: true })
+  @Field({ nullable: true })
+  offeredTo?: string;
+
+  @Column()
+  @Field()
+  timestamp: string;
+
+  @OneToOne(
+    () => DIDEntity,
+    did => did.id,
+    { eager: true },
```

Comment:
Why this `eager: true` here?

Reply:
I thought it would be better to automatically resolve this relation, cause we always want to see the didDoc of the asset.